New Document: spinbutton role

estelle opened the pull request:
- [X] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

estelle committed:
DRAFt: New Document: spinbutton role

estelle committed:
DRAFt: New Document: spinbutton role

dipikabh commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+
+HTML's `<input type="number">` implicitly has the `role` of `spinbutton`. HTML's `<input type="date">` has 3 nested spin buttons; one each for month, day, and year. When using semantic HTML form elements for their intended purposes, do not use `aria-valuemax` or `aria-valuemin` attributes; use `min` and `max` instead. Otherwise, any global aria-* attributes and any other aria-* attributes are app ...
+
+### Prefer HTML
```
```suggestion
### Prefer semantic HTML
```

dipikabh commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+
+An accessible name is **required**. If the spinbuttons's role is applied to an HTML {{HTMLElement('input')}} element, the accessible name can come from the associated {{HTMLElement('label')}}. Otherwise use [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) if a visible label is present or  [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) if ...
+
+When not using the HTML {{HTMLElement('input')}} element to create your spinbutton, include the {{htmlattrxref('tabindex')}} attribute to make the spinbutton focusable. `spinbutton` is user-interactive and therefore requires being able to receive focus. Focus should be placed on the spinbutton input, and not on the associated buttons that increment and decrement the `spinbutton` value.
```
```suggestion
When not using the HTML {{HTMLElement('input')}} element to create your spinbutton, include the {{htmlattrxref('tabindex')}} attribute to make the spinbutton focusable. The `spinbutton` role is user-interactive, and therefore, requires being able to receive focus. Focus should be placed on the spinbutton input and not on the associated buttons that increment and decrement the spinbutton value.
```

dipikabh commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+
+When `aria-valuetext` is an important feature for a spinbutton, consider using {{HTMLElement('select')}} with {{HTMLElement('option')}} elements instead. The `<select>` doesn't provide for a  every option's value is more accessible to all users, not just users of assistive technology.
+
+An accessible name is **required**. If the spinbuttons's role is applied to an HTML {{HTMLElement('input')}} element, the accessible name can come from the associated {{HTMLElement('label')}}. Otherwise use [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) if a visible label is present or  [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) if ...
```
```suggestion
An accessible name is **required**. If the `spinbutton` role is applied to an HTML {{HTMLElement('input')}} element, the accessible name can come from the associated {{HTMLElement('label')}}. Otherwise, use [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) if a visible label is present or  [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) if a visible label is not present.
```

dipikabh commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+
+The `aria-valuetext` value must be updated as the `value` or `aria-valuenow` is updated. While there is no equivalent HTML attribute for `<input type="range">`, you can include `aria-valuetext` on any {{htmlelement('input')}} type. ARIA attributes are supported on semantic HTML elements.
+
+When `aria-valuetext` is an important feature for a spinbutton, consider using {{HTMLElement('select')}} with {{HTMLElement('option')}} elements instead. The `<select>` doesn't provide for a  every option's value is more accessible to all users, not just users of assistive technology.
```
The second sentence is not clear.

estelle replied:
no kidding! what was I trying to say?

dipikabh commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+
+The optional [`aria-valuetext`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-valuetext) attribute is included when the `aria-valuenow` numeric value doesn't reflect the intended value of the spinbutton.  As the optional minimum, maximum, and current values are all numeric, when the values those numbers represent are not numeric, the `aria-valuetext` attribute should be included with a strin ...
+
+The `aria-valuetext` value must be updated as the `value` or `aria-valuenow` is updated. While there is no equivalent HTML attribute for `<input type="range">`, you can include `aria-valuetext` on any {{htmlelement('input')}} type. ARIA attributes are supported on semantic HTML elements.
```
```suggestion
The `aria-valuetext` value must be updated as the value or `aria-valuenow` is updated. While there is no equivalent HTML attribute for `<input type="range">`, you can include `aria-valuetext` on any {{htmlelement('input')}} type. ARIA attributes are supported on semantic HTML elements.
```

dipikabh reviewed (changes requested):
Hi @estelle, I've done a readthrough of this document - just a readability and language check. I hope my suggestions have not introduced any technical inaccuracy - please feel free to redo the suggestion if it is not accurate.

estelle commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+- [`aria-valuenow`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-valuenow) (required)
+  - : Set to a decimal value between `aria-valuemin` and `aria-valuemax`, indicating the current value of the spinbutton. If not present, there is no default value.
+- [`aria-valuetext`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-valuetext)
+  - : Assistive technologies often present the value of `aria-valuenow` as a number. If this would not be accurate, use `aria-valuetext` to provide the spinbutton with a more understandable value.
```
this would change the meaning.

estelle committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

estelle committed:
Update files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

estelle committed:
edits per review

OnkarRuikar committed:
Update redirected URLs from `*.apple.com` (#15141)

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 13) (#15142)

* update method pages

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

* added 'return value' section

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 14) (#15143)

* update method pages

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

* removed `{ }` from the param

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

alattalatta committed:
Remove indeterminate from Additional attributes (#15152)

abdirahmanjama committed:
refactor: keep tags lowercase (#15116)

* refactor: keep tags lowercase

What? 
- Update content to highlight the typical convention of keeping tags lowercase. 

Why?
- Make readers aware of best practices from the get-go. 
- Enable them to inhibit this habit early on in their learning journey.

* feat: update files/en-us/web/html/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

teoli2003 committed:
Remove link to old project Midas (#15150)

Fixes #15149

OnkarRuikar committed:
Client-side_javascript_frameworks: fixed typos (#15064)

* Client-side_javascript_frameworks: fixed typos

* Update files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/introduction/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

protofarer committed:
Update obsolete information (#15154)

- ClusterWS has been deprecated, see: https://github.com/ClusterWS/ClusterWS
- CWS has been deprecated, see: https://github.com/ClusterWS/cWS

benjsmyth committed:
Fix title inconsistency (#15156)

- The label "Content Categories" is inconsistent with the rest of the labels on this page
- It should be corrected to "Content categories"

dependabot[bot] committed:
Bump @mdn/yari from 0.11.6 to 0.11.18 (#15158)

Bumps [@mdn/yari](https://github.com/mdn/yari) from 0.11.6 to 0.11.18.
- [Release notes](https://github.com/mdn/yari/releases)
- [Changelog](https://github.com/mdn/yari/blob/main/docs/npm-releases.md)
- [Commits](https://github.com/mdn/yari/compare/v0.11.6...v0.11.18)

---
updated-dependencies:
- dependency-name: "@mdn/yari"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

NiedziolkaMichal committed:
translate & scale interactive examples (#15151)

mfuji09 committed:
Replace {{event}} with {{domxref}} in <video> (#15163)

codersjj committed:
Fix type in "Iteration protocols" doc (#15164)

brandhawa99 committed:
Luxon docs link is broken (#15173)

* Luxon docs link is broken

In the see also section the Lexon link takes you to a file not found page and not the actual Luxon docs.

* Update index.md

erodrigufer committed:
Fix incorrect grammar, small change (#15175)

A verb was missing.

MendyBerger committed:
Set wasm interactive example size (#15178)

MrBrain295 committed:
HTTP -> HTTPS (#15177)

sideshowbarker committed:
Document that setTimeout coerce non-number timeout values into numbers (#15185)

Fixes https://github.com/mdn/content/issues/15129

OnkarRuikar committed:
fix typos (#15179)

OnkarRuikar committed:
update redirected URLs (#15180)

OnkarRuikar committed:
Update redirected URLs (#15181)

cnjhb committed:
Remove redundant list elements (#15187)

teoli2003 committed:
Add info that these exceptions are DOMException (#15146)

yin1999 committed:
remove deprecated `{{Interface}}` macro (#15188)

sideshowbarker committed:
Fix typo in earlier commit, and make one wording refinement (#15189)

A typo crept into 32a7108 (https://github.com/mdn/content/pull/15185) —
so this change fixes that, as well as changing another word to be more
precise.

sideshowbarker committed:
Update Intl.ListFormat() params section + add more examples (#15191)

Fixes https://github.com/mdn/content/issues/15167

This drops the statement “When `style` is "`short`" or "`narrow`",
"`unit`" is the only allowed value for the type option” — which doesn’t
align with the spec nor with implementations (that wording/restriction
was in the original proposal but ultimately didn’t make its way into the
spec. This also adds a lot more examples.

OnkarRuikar committed:
Remove repeated words (part 2) (#15184)

* Remove repeated words (part 2)

* repeated punctuations

* more repeated punctuations

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 15) (#15182)

* update method pages

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

* moved param section after syntax

* put param section after syntax

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Elchi3 committed:
Update popstate event docs (#15192)

* Update popstate event docs

* Fix more onpopstate links

dependabot[bot] committed:
Bump @mdn/yari from 0.11.18 to 0.11.22 (#15200)

Bumps [@mdn/yari](https://github.com/mdn/yari) from 0.11.18 to 0.11.22.
- [Release notes](https://github.com/mdn/yari/releases)
- [Changelog](https://github.com/mdn/yari/blob/main/docs/npm-releases.md)
- [Commits](https://github.com/mdn/yari/compare/v0.11.18...v0.11.22)

---
updated-dependencies:
- dependency-name: "@mdn/yari"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gpaumier committed:
Fix links to fit-content() (#15199)

* fix link to fit-content() in grid-template-columns

* fix link to fit-content() in grid-template-rows

* fix link to fit-content() in grid-auto-columns

* fix link to fit-content() in grid-auto-rows

* Fix link to fit-content() in FF exp features

* Fix link to fit-content() in css_grid_layout

simonvarey committed:
Removing unneeded underscores (#15204)

This page had some unneeded underscores. I removed them.

NiedziolkaMichal committed:
Removed Experimental flags (#15207)

b0nbon1 committed:
fix Element.shadow example bug (#13490)

jespertheend committed:
Clarify the return type of Cache.keys() (#15206)

Elchi3 committed:
Remove Selection.selectionLanguageChange page (#15198)

aardrian committed:
Some clean-up (#14704)

* Some clean-up

Corrections for `aria-details`, `aria-description`; clarity in where/which/when/how.

* Update files/en-us/web/accessibility/aria/annotations/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

wbamberg committed:
Remove tables in shorthand property page (#15213)

wbamberg committed:
Modernize wordcount example (#15209)

dipikabh committed:
Add @layer syntax and examples to @import (#15205)

Co-authored-by: Dipika Bhattacharya <dbhattacharya@mozilla.com>

Elchi3 committed:
Update storage event docs (#15202)

wbamberg committed:
Remove scrollbar-width and scrollbar-color interactive examples (#15174)

* Remove scrollbar-width interactive example

* Also scrollbar-color

NiedziolkaMichal committed:
More info in datalist element (#14834)

* More info in datalist element

* Rewritten date & time example

* Fixed height of color & range examples

* Update files/en-us/web/html/element/datalist/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

Elchi3 committed:
Update pageshow/pagehide events to new structure (#15195)

Elchi3 committed:
Update languagechange event docs (#15197)

NiedziolkaMichal committed:
Multiple values in animation-iteration-count (#15171)

MathiasWP committed:
Add case that affects identifying containing block (#15088)

Spent a lot of time debugging a weird containing-block case, and it was caused by an ancestor having the backdrop-filter property (in chromium).

nschonni committed:
fix: various typos (#15219)

hamishwillee committed:
tolocaledatestring: Add more obvious cross linking to options (#15216)

wbamberg committed:
enterkeyhint is supported (#15220)

* enterkeyhint is supported

* Un-recommend mozactionhint some more

OnkarRuikar committed:
update redirected url (#15224)

wbamberg committed:
Add triage process doc (#15125)

* Add triage process doc

* Add needs-info

* fix typo

* Update with review feedback

* Remove quorum bit

tomayac committed:
Typo fix (#15229)

Elchi3 committed:
Fix offline/online event docs (#15228)

lukewarlow committed:
Remove experimental header box from permissions API and permissions query (#15231)

mpuels committed:
Fix typo (#15239)

juansalvatore committed:
Update index.md (#15243)

Fix typo

Elchi3 committed:
Update load events to new format (#15237)

lukewarlow committed:
Remove experimental header box from Intl.supportedValuesOf (#15234)

lukewarlow committed:
Remove experimental header box from Screen Orientation API and Screen Orientation unlock (#15233)

lukewarlow committed:
Remove experimental header box from WritableStreamDefaultController pages (#15236)

lukewarlow committed:
Remove experimental header box from IntersectionObserverEntry pages (#15238)

Elchi3 committed:
Fix promise rejection events (#15244)

sideshowbarker committed:
Deprecate the param element and HTMLParamElement API (#15225)

https://github.com/whatwg/html/commit/04a11f2
(https://github.com/whatwg/html/pull/7816) obsoletes the HTML “param”
element and the HTMLParamElement API

Related BCD change: https://github.com/mdn/browser-compat-data/pull/15955

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 16) (#15223)

* update method pages

* update method pages

* handled optional param

* Update files/en-us/web/api/selection/containsnode/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

* address review comment

* added space

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Elchi3 committed:
Remove the WindowEventHandlers mixin (#15246)

dependabot[bot] committed:
Bump @mdn/yari from 0.11.22 to 0.12.0 (#15247)

Bumps [@mdn/yari](https://github.com/mdn/yari) from 0.11.22 to 0.12.0.
- [Release notes](https://github.com/mdn/yari/releases)
- [Changelog](https://github.com/mdn/yari/blob/main/docs/npm-releases.md)
- [Commits](https://github.com/mdn/yari/compare/v0.11.22...v0.12.0)

---
updated-dependencies:
- dependency-name: "@mdn/yari"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

stefanhamburger committed:
Fix typo (#15248)

mfuji09 committed:
replace {{event}} with {{domxref}} for cut/copy/paste events (#15250)

mfuji09 committed:
These events aren't experimental (#15249)

NiedziolkaMichal committed:
Interactive example in counter-set (#15240)

wbamberg committed:
Clarify global attributes for input elements (#15218)

axis68 committed:
Update formatting React Getting Started (#15235)

Because of a missing <p> tag, both rows have a different formatting

de-oz committed:
Typo fix (#15255)

thank you

mfuji09 committed:
Replace {{event}} macros (#15254)

rmedaer committed:
Fix typo: "your" instead of "you" (#15257)

teoli2003 committed:
Change for plural (#15256)

Fixes #15252

wbamberg committed:
Rename html->md

wbamberg committed:
Convert contents

NiedziolkaMichal committed:
More values in object-position syntax (#15253)

drj11 committed:
Fix assertion about abbreviated S command (#15251)

teoli2003 committed:
Amend API method template to describe `DOMException` usage (#15221)

* Amend API method template to describe `DOMException` usage

* Typo fix

* Update files/en-us/mdn/structures/page_types/api_method_subpage_template/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

Co-authored-by: wbamberg <will@bootbonnet.ca>

NiedziolkaMichal committed:
page-orientation descriptor in page at-rule (#15261)

* page-orientation descriptor in page at-rule

* Reorder

teoli2003 committed:
Create hexadecimal color syntax page (#14813)

* Create hexadecimal color syntax page

* Add missing link and ...

* Fix dl

* Apply grammar suggestions from code review

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Fix structure (Value as an H3)

* Remove hidden class for live sampe

Co-authored-by: wbamberg <will@bootbonnet.ca>

pinksynth committed:
Fix typo in example (#15263)

Change `</li>` to `</div>` in example.

giorgitsiklauri committed:
Correct tag omission for <tbody> (#11912)

* Update index.md

Clerical error fixed ("Must not be present" should have been "must be present").

* added Teoli's suggestion

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

OnkarRuikar committed:
Fix typos (#15270)

* Fix typos

* Update files/en-us/learn/css/building_blocks/values_and_units/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

NiedziolkaMichal committed:
Defined left/right page in page-break properties (#15262)

mfuji09 committed:
Remove touchenter and touchleave events from HTMLElement interface page (#15265)

mfuji09 committed:
Replace {{event}}s with {{domxref}}s for `pointerdown`, `pointerout` and `pointerleave` (#15266)

teoli2003 committed:
Add color wheel to the glossary (#14734)

* Add color wheel in the glossary

* Compresses added image

* Fixing image

* Fix links (missing /docs/)

* Add link to <color> in See also

* Link from other articles

* suggested edits

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

computersarecool committed:
Clarifies rules around sectioning content (#15274)

Sheraff committed:
Fix explanation of aria-live=off (#15273)

aria-live=off should be "do NOT inform on every update" since that's what it does.

glmvc committed:
Fix Issue #13671: Make sections consistent across modules (#15054)

* Fix Issue #12671: Make sections consistent across modules

* Update files/en-us/learn/css/building_blocks/debugging_css/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/learn/css/building_blocks/images_media_form_elements/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/learn/css/css_layout/responsive_design/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/learn/html/introduction_to_html/document_and_website_structure/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/learn/html/tables/advanced/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/learn/html/tables/advanced/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Adjustments to the requested changes

* Copy edit

* Update files/en-us/learn/css/building_blocks/selectors/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/en-us/learn/css/building_blocks/selectors/pseudo-classes_and_pseudo-elements/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

Co-authored-by: wbamberg <will@bootbonnet.ca>

MrBrain295 committed:
Remove useless link (#15279)

* Remove useless link

* Update index.md

yoshinorin committed:
fix typo in `glossary/deep_copy` (#15276)

mi-v committed:
Formatting fix in regex guide (#15278)

A single backslash get eaten at the MDN website as it's probably considered escaping the following curly brace.

NiedziolkaMichal committed:
Interactive Example in margin-block, margin-inline & ruby-position (#15277)

queengooborg committed:
Remove MediaTrack[Supported]Constraints.cursor from MDN (#15293)

mehdiakiki committed:
Fix typo (#15290)

abergquist committed:
Update index.md (#15286)

queengooborg committed:
Remove CanvasRenderingContext.currentTransform from MDN (#15294)

mfuji09 committed:
Use “pressed” instead of “depressed” (#15298)

OnkarRuikar committed:
Update redirected URLs from http -> https (part 21) (#15301)

OnkarRuikar committed:
Update redirected URLs from `*.oculus.com` (part 20) (#15299)

sideshowbarker committed:
Fix broken links in “Your first form” article (#15305)

Fixes https://github.com/mdn/content/issues/15280

OnkarRuikar committed:
Fix typos (#15303)

* Fix typos

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

* Update files/en-us/web/javascript/reference/errors/invalid_right_hand_side_instanceof_operand/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

sideshowbarker committed:
Add RTCDataChannel to list of transferable objects (#15306)

Fixes https://github.com/mdn/content/issues/15242

See https://w3c.github.io/webrtc-extensions/#rtcdatachannel-extensions

OnkarRuikar committed:
Update redirected URLs from `https` -> `http` (part 18) (#15267)

* Update redirected URLs from `https` -> `http`

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 17) (#15268)

* Update methdo pages

* corrected the param name

* In param names, Ns -> NS

nschonni committed:
fix: various typos (#15260)

* fix: various typos

* fix: duplicate words

* Update files/en-us/web/api/securitypolicyviolationevent/statuscode/index.md

Co-authored-by: Onkar Ruikar <87750369+OnkarRuikar@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>
Co-authored-by: Onkar Ruikar <87750369+OnkarRuikar@users.noreply.github.com>

OnkarRuikar committed:
Fix typos  (#15302)

* Fix typos

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

OnkarRuikar committed:
Honor product/company name cases - nitpicking (part 7) (#15304)

* Honor product/company name cases (part 7)

* Update files/en-us/web/api/network_information_api/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

NiedziolkaMichal committed:
Deprecated viewport at-rule (#15289)

* Deprecated viewport at-rule

* document, property, scroll-timeline

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 18) (#15300)

* Update method pages

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

* added ### Parameters title

* Added ### Parameters title

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

sideshowbarker committed:
Add arguments object & Intl.Segments to iterables list; note NodeList (#15312)

Fixes https://github.com/mdn/content/issues/15203

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segments/@@iterator
and http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Functions/arguments/@@iterator
and see https://dom.spec.whatwg.org/#interface-nodelist and
https://webidl.spec.whatwg.org/#idl-iterable (the WebIDL “iterable”
notation maps to the ES @@iterator property).

OnkarRuikar committed:
Update redirected URLs from `http` -> `https` (part 19) (#15269)

* Update redirected URLs from `http` -> `https`

* removed specific part of the link

teoli2003 committed:
Add DOMException (in WebSocket.*) (#15313)

sideshowbarker committed:
Note that Map.p.set keys & values can be any JS data types (#15314)

Fixes https://github.com/mdn/content/issues/15287

mfuji09 committed:
Remove unnecessary keyword (#15319)

mfuji09 committed:
Replace `Boolean` with boolean (primitive) in 'CloseEvent' (#15327)

mfuji09 committed:
Fix the sidebar in `NDEFReadingEvent` (#15323)

mfuji09 committed:
Fix various typos in CloseEvent (#15328)

imphil committed:
Remove mention about "future" Quantum CSS updates (#15317)

Quantum CSS has been in Firefox releases for a fair while now. Remove the notes that talk about it as future technology.

paulhicks91 committed:
Removed hidden from "passwordrules" div (#14314)

When copying the html into an online sandbox, the tooltip would not pop up when hidden was is in the "passwordrules" div. Taking it out allowed the tooltip to appear on hover and disappear when not hovering. Not sure if it's a browser-specific issue, I'm using Chrome Version 99.0.4844.83 (Official Build) (x86_64) on macOS Version 12.2.1.

gardens-of-lorien committed:
Past tense of to string is strung (#15332)

This dictionary.com https://www.dictionary.com/browse/string entry lists strung as the preferred past tense of the verb to string.

vch-7 committed:
Fixed the If-None-Match header reference in the Cache update section (#15324)

* Fixed the `If-None-Match` header reference in the `Cache update` section

* Minor grammatical fix

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

vch-7 committed:
Fixed the explanation of the `Vary` header (#15333)

mfuji09 committed:
Remove links to `DocumentAndElementEventHandlers` and `TouchEventHand… (#15320)

teoli2003 committed:
Modernize the structure of TimeRanges.start()/end() (#15316)

queengooborg committed:
Remove intrinsicsize <img> HTML attribute (#15321)

queengooborg committed:
Remove scoped <style> HTML attribute (#15322)

Mike96Angelo committed:
Documenting some regex group limitations (#15159)

alabhyajindal committed:
Fixed a typo (#15341)

OnkarRuikar committed:
Remove broken links (#15339)

AnilSeervi committed:
http -> https (#15342)

https://github.com/mdn/content/pull/15339#pullrequestreview-951361916

OnkarRuikar committed:
Replace unwanted URL with a live sample (#15338)

OnkarRuikar committed:
Fix spelling errors (#15336)

nschonni committed:
fix: Markdownlint formatting (#15330)

* fix: Markdownlint formatting

- Trailing spaces
- Empty list items
- List indentation

* fix: headings in lists

OnkarRuikar committed:
Update redirected URLs from `http` -> `https` (part 22) (#15334)

* Update redirected urls from `http` to `https`

* Update the method page

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 19) (#15335)

* Update method pages

* Update files/en-us/web/api/xpathresult/snapshotitem/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

myshov committed:
Replace lacuna vaule with initial value (#15344)

alattalatta committed:
Remove non-standard extensions from Document (#15347)

simonvarey committed:
Removing unneeded underscores (#15353)

This page had some unneeded underscores. I removed the unneeded underscores on this page.

dependabot[bot] committed:
Bump @mdn/yari from 0.12.0 to 0.13.1 (#15351)

Bumps [@mdn/yari](https://github.com/mdn/yari) from 0.12.0 to 0.13.1.
- [Release notes](https://github.com/mdn/yari/releases)
- [Changelog](https://github.com/mdn/yari/blob/main/docs/npm-releases.md)
- [Commits](https://github.com/mdn/yari/compare/v0.12.0...v0.13.1)

---
updated-dependencies:
- dependency-name: "@mdn/yari"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

MrBrain295 committed:
Replace link (#15354)

myshov committed:
Fix nuance in behavior of alpha component of :visited styles in Firefox (#15346)

kdagnan committed:
Update Ember Guide with Array Tracking (#14798)

nschonni committed:
fix: various typos and formatting (#15356)

mfuji09 committed:
Fix a link flaw in NDEFRecord (#15364)

mfuji09 committed:
Fix a sidebar (#15363)

korostelevm committed:
Add reference to Promise.allSettled (#15357)

mousetail committed:
Optional Chaining: Add intros to currently context-less examples (#15352)

Commit:
typo (#15345)

mseeley committed:
Update yield docs (#15271)

jopemachine committed:
Fix AbortSignal»throwIfAborted example code (#15368)

OnkarRuikar committed:
Fix typo (#15372)

OnkarRuikar committed:
Fix typos (#15371)

OnkarRuikar committed:
Update redirected URLs to and from `github.io`` (#15369)

OnkarRuikar committed:
Correct spellings (#15367)

OnkarRuikar committed:
Fix a typo (#15370)

* Fix a typo

* Update files/en-us/web/api/service_worker_api/using_service_workers/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

OnkarRuikar committed:
Honor product/company name cases - nitpicking (part 8) (#15337)

* Honor product/company name cases

* Update files/en-us/web/accessibility/aria/web_applications_and_aria_faq/index.html

* Update files/en-us/web/mathml/fonts/index.md

* Update remaining `OS X` to `macOS`

* updating as per review suggestions

* Apply suggestions from code review

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 20)  (#15365)

* Update method pages

* Update files/en-us/web/api/writablestreamdefaultwriter/abort/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

OnkarRuikar committed:
URLs: Prefer `https` over `http` (part 23) (#15366)

* URLs: Prefer `https` over `http`

* Remove unwanted url

OnkarRuikar committed:
Update moved URLs with their new locations (part 1) (#14738)

* mdn redirects

* Small corrections

* small correction

grandeljay committed:
Fix missing "if" (#15379)

dipikabh committed:
Add new pages for dynamic-range and video-dynamic-range and update media pages (#15375)

Co-authored-by: Dipika Bhattacharya <dbhattacharya@mozilla.com>

teoli2003 committed:
Remove prototype + mention of dictionaries (#15382)

dependabot[bot] committed:
Bump @mdn/yari from 0.13.1 to 0.14.1 (#15387)

Bumps [@mdn/yari](https://github.com/mdn/yari) from 0.13.1 to 0.14.1.
- [Release notes](https://github.com/mdn/yari/releases)
- [Changelog](https://github.com/mdn/yari/blob/main/docs/npm-releases.md)
- [Commits](https://github.com/mdn/yari/compare/v0.13.1...v0.14.1)

---
updated-dependencies:
- dependency-name: "@mdn/yari"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dipikabh committed:
Add CSS media features to FF 100 release note (#15378)

Co-authored-by: Dipika Bhattacharya <dbhattacharya@mozilla.com>

codersjj committed:
Add more APIs in "Built-in APIs accepting iterables" (#15383)

alattalatta committed:
Make syntax sections of group() and groupEnd() comply with guideline (#15118)

* Make syntax sections of group() and groupEnd() comply with guideline

* Apply the new guideline for return value sections

jeffposnick committed:
Use HTML in no-markdown tables for crossorigin docs (#15389)

* Use HTML, not Markdown, in no-markdown tables

* Re-add /en-US

teoli2003 committed:
Adapt SubtleCrypto pages to modern structure (#15388)

* Adapt pages to modern structure

* Fix decrypt()

* Fix typo

* Fix typos

* Fix dl

* Fix typo

teoli2003 committed:
Linkify TypeError (#15377)

myshov committed:
Fix description of string value in grid-template-areas (#15393)

bappyasif committed:
fixed a negligible typo (#15399)

'period / full-stop' was within 'brackets', in that statement, and i just moved it outside brackets, thanks :)

de-oz committed:
Correcting grammatical error (#15405)

nschonni committed:
fix: Markdownlint issues (#15403)

wbamberg committed:
Fix 15329: link to CSP header page for policy directives (#15406)

trace-mark committed:
Link formatting makes one symbol look like another (#15157)

In the case of greater than, less than, and plus operators the link underlining makes the symbol look like another symbol with a different meaning (greater than looks like greater than or equal to, less than looks like less than or equal to, and plus looks like plus/minus). Placing the word after the symbol may help disambiguate the presentation.

zhangenming committed:
Better readability(Personal opinion). (#15380)

* Better readability(Personal opinion).

* Apply suggestions from code review

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

wbamberg committed:
Fix example heights in the gap page (#15402)

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 21)  (#15415)

* Update method pages

* Apply suggestions from code review

* small correction

OnkarRuikar committed:
Update redirected URLs happening due to subdomain changes (#15417)

OnkarRuikar committed:
Update redirected URLs happening due to subdomain change (part 25) (#15416)

* Update redirected URLs

* Update redirected URLs happening due to subdomain changes

* small correction

* small correction

* small correction

* small correction

teoli2003 committed:
Update return value text when there is no returned value (#15145)

* Update return value text when there is no returned value.

* A bit better

* Update files/en-us/mdn/structures/syntax_sections/index.md

Co-authored-by: Onkar Ruikar <87750369+OnkarRuikar@users.noreply.github.com>

* Make the Parameters section mandatory

* Update files/en-us/mdn/structures/syntax_sections/index.md

Co-authored-by: Onkar Ruikar <87750369+OnkarRuikar@users.noreply.github.com>

Co-authored-by: Onkar Ruikar <87750369+OnkarRuikar@users.noreply.github.com>

OnkarRuikar committed:
Update redirected URLs happening due to subdomain changes (#15418)

OnkarRuikar committed:
Update redirected URLs happening due to subdomain changes (#15420)

OnkarRuikar committed:
Honor product names (#15422)

OnkarRuikar committed:
Google_Chrome: Update outdated link (#15423)

teoli2003 committed:
Fix exception example (#15424)

* Fix exception example

The original exception was a JS exception (a _simple_ exception per the WebIDL spec) while we are demonstrating the second type of exception, a `DOMException`.

* Fix comment

dependabot[bot] committed:
Bump @mdn/yari from 0.14.1 to 0.14.3 (#15432)

Bumps [@mdn/yari](https://github.com/mdn/yari) from 0.14.1 to 0.14.3.
- [Release notes](https://github.com/mdn/yari/releases)
- [Changelog](https://github.com/mdn/yari/blob/main/docs/npm-releases.md)
- [Commits](https://github.com/mdn/yari/compare/v0.14.1...v0.14.3)

---
updated-dependencies:
- dependency-name: "@mdn/yari"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

yin1999 committed:
fix: resolve errors in links (#15431)

yin1999 committed:
fix: correct the size of first packet of TCP (#15426)

The size of first packet of TCP should be about 14KB not 14Kb.
Also fix the broken link to `Abstract Syntax Tree`.

mikedubcurry committed:
or => nor (#15430)

just a grammatical nit pick, use nor for exclusionary clauses. Either way, I think the meaning of content is clear

meatnordrink committed:
Remove outdated information about `xlink:href` (#15435)

* Remove outdated information about `xlink:href`

This warning suggests that `xlink:href` may still be necessary for cross-browser compatibility; however, the table at the bottom shows that the stable version of all browsers (including IE) now support `href`, so this warning is now misleading. In addition, it conflicts with the more accurate warning further down the page that `xlink:href` is deprecated.

* Restoring a portion of the warning

This may still be redundant with the warning below, but I restored it as it could be useful to anyone familiar with the old `xlink:href` pattern.

estelle committed:
<menu>: removed experimental language (#15412)

* <menu>: removed experimental language

The `<menu>` element was implemented in IE6, so I don't think it's experimental anymore.
Changed the language for context menu since that was removed a LONG time ago, so gave it wording to make it seem older and less of an issue needing fixing.

* put {{HTMLRef}} back in

bappyasif committed:
attempting to fix what seems to me a probable typo (#15436)

live example is 'below' that section (Steps to complete) not 'above', and this commit has that change in it

bappyasif committed:
brought changes to article macro reference for next and previous navi… (#15401)

* brought changes to article macro reference for next and previous navigation

* also added macro references at top of those two articles as well

NiedziolkaMichal committed:
Interactive examples in inset-*, aspect-ratio, user-select, scroll-margin-inline (#15428)

YangFong committed:
docs: adjust the indentation (#15425)

lex111 committed:
Expand live sample height on grid-auto-flow page (#15427)

ravage84 committed:
"prepended to" instead of "appended to the beginning of" (#15437)

teoli2003 committed:
Remove Options subsections in Parameters (#15419)

wbamberg committed:
Fix for 13925 (#15439)

wbamberg committed:
Fix issue 15359 (#15440)

estelle committed:
DL layout fix

estelle committed:
DL layout fix

estelle committed:
sentence simplification

estelle committed:
sentence simplification

estelle committed:
clarifications

estelle committed:
clarifications

estelle committed:
live example

estelle committed:
live example

hamishwillee committed:
FF100 RelNote: WebAssembly Exceptions (#15410)

Konrud committed:
Fix example markup (#15386)

Having a native HTML button makes the use of these attributes redundant and confusing.

yin1999 committed:
Update index.md (#15444)

myshov committed:
Add interpolation page to glossary (#15281)

* Add interpolation page to glossary

* Add links to interpolation page

* Update files/en-us/glossary/interpolation/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

OnkarRuikar committed:
Update redirected URLs happening due to subdomain changes (#15449)

OnkarRuikar committed:
Fix spelling errors (part 3) (#15448)

* Correct spellings

* Update files/en-us/games/techniques/3d_on_the_web/basic_theory/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

ggorlen committed:
Clarify "ray casting" rather than "ray tracing" (#15442)

The original text, "ray tracing" isn't as accurate as "ray casting", which is the primary style of rendering shown in the tutorial.

OnkarRuikar committed:
Syntax_sections: fix escape characters (#15450)

OnkarRuikar committed:
Update redirected URLs due to domain change (part 29) (#15451)

* Update redirected URLs happening due to subdomain changes

* Update files/en-us/learn/common_questions/how_much_does_it_cost/index.md

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

teoli2003 committed:
Migrate Exceptions section to modern structure (#15421)

* Migrate Exceptions section to modern structure

* Update files/en-us/web/api/htmltableelement/deleterow/index.md

Yes, that's much better.

Co-authored-by: wbamberg <will@bootbonnet.ca>

Co-authored-by: wbamberg <will@bootbonnet.ca>

wbamberg committed:
Add an example for SpeechSynthesisUtterance.rate (#15453)

teoli2003 committed:
RangeError and TypeError are JS exception; SyntaxError normally not (#15438)

* RangeError and TypeError are JS exception; SyntaxError normally not

* Fixes grammar

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Revert thrown -> returned for promises

* Simplify sentence

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Revert thrown -> reverted for a promise

* Fix grammer

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Remove useless ifs

Co-authored-by: wbamberg <will@bootbonnet.ca>

Co-authored-by: wbamberg <will@bootbonnet.ca>

yin1999 committed:
fix livesample in HTMLSelectElement.selectedIndex (#15457)

teoli2003 committed:
Fix broken interactive example (#15471)

teoli2003 committed:
Add code markings to parameters and exceptions. (#15474)

OnkarRuikar committed:
Update method pages (#15452)

OnkarRuikar committed:
WebAPI: Update Method pages to modern structure (part 22) (#15447)

* Update method pages

* Update remaining pages

* start index at 1

* Apply suggestions from code review

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

* Start index at 1

* Use unicode `…` instead of 3 dots.

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

dependabot[bot] committed:
Bump @mdn/yari from 0.14.3 to 0.14.16 (#15478)

Bumps [@mdn/yari](https://github.com/mdn/yari) from 0.14.3 to 0.14.16.
- [Release notes](https://github.com/mdn/yari/releases)
- [Changelog](https://github.com/mdn/yari/blob/main/docs/npm-releases.md)
- [Commits](https://github.com/mdn/yari/compare/v0.14.3...v0.14.16)

---
updated-dependencies:
- dependency-name: "@mdn/yari"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

estelle committed:
Cascade layer: clarifications (#15443)

* lowered reading level ("come before" instead of "precede")
* put information on anonymous cascade layers in one section and expanded on the definition.

wbamberg committed:
Fix issue 14431: Document alternative to CSSRule.type (#15454)

* Fix issue 14431: Document alternative to CSSRule.type

* Link to constructor.name

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

Co-authored-by: Jean-Yves Perrier <jypenator@gmail.com>

AJ-Kandra committed:
Fix typo "centre" to "center" (#15482)

hrldcpr committed:
fix typo (#15484)

AnilSeervi committed:
fix active classname selector (#15472)

butlermatt committed:
300 Status code message should be pluralized (#15485)

saadahmadhere committed:
[fix] typo in index.md (#15481)

change args1 to args0

teoli2003 committed:
Fixes argument names (#15483)

Fixes #15479

dipikabh commented:
Hi @estelle, I think these comments are pending. Everything else looks good.

- https://github.com/mdn/content/pull/14764#discussion_r853027513 (suggestion for changing heading to "Comparing the spinbutton role with other range options"  to avoid "The" in the beginning of the heading)
- https://github.com/mdn/content/pull/14764#discussion_r853068909 - I've added a new suggestion/question
- https://github.com/mdn/content/pull/14764#discussion_r853075053 - can we rewrite the description for `aria-label` as:
Defines a string value that labels the spinbutton element. This provides an accessible name to the element when no visible label is available.
- https://github.com/mdn/content/pull/14764#discussion_r853089814
- https://github.com/mdn/content/pull/14764#discussion_r853101520

estelle commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+
+> **Note:** It is recommended to use a native [`<input type="number">`](/en-US/docs/Web/HTML/Element/input/number) (or the date and time input types for dates and time) element rather than the `spinbutton` role. User agents provide a stylized widget for the number, date, and time input elements which include buttons to increment and decrement the values, limiting the range of values if minimum an ...
+
+### The `spinbutton` role compared other range options
```
Changed to "ARIA range options" as it's not a comparison but an intro, so compare wasn't really the right word in the first place

estelle committed:
Update files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

estelle commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+
+> **Warning** To change the spinbutton value, touch-based assistive technologies need to respond to user gestures for increasing and decreasing the value by synthesizing key events. Fully test spinbutton widgets using assistive technologies on devices where touch is a primary input mechanism before using the `spinbutton` role (and all range widgets).
+
+#### Common attributes
```
It is a subsection of the range options section, which is a level 3 header.

estelle committed:
edits per suggestions

estelle committed:
argh

estelle commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+| Right and Up arrows | Increase the selected value by one step |
+| Left and Down arrows | Decrease the selected value by one step|
+| Page Up | (Optional) increase the value by a set amount greater than or equal to one step |
+| Page Up and Page Down | (Optional) decrease the value by a set amount greater than or equal to one step |
```
yes! good catch.

estelle commented on files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:
```diff
@@ -0,0 +1,172 @@
+- [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby)
+  - : Defines the string value or identifies the element (or elements) that label the spinbutton element providing an accessible name. An accessible name is required.
+- [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) 
+ - : Defines a string value that label the spinbutton element providing an accessible name when no visible label is available to provide the required accessible name.
```
It now reads " Defines a string value providing the accessible name when no visible label is available to provide the required accessible name via {{HTMLElement('label')}} or `aria-labelledby`."

estelle committed:
saving before committing is more effective

estelle commented:
rebase failed. Closing and opening a new PR
